### ISSUE TITLE
[ENHANCEMENT] Install extra so `camel_tools.utils` can be used without torch

### ISSUE THREAD

**Issue by shaniaakhan21** [CAMeL-Lab/camel_tools#173]
**What feature would you like to improve?**

Packaging / install dependencies for `camel_tools.utils` (specifically
`camel_tools.utils.dediac` and `camel_tools.utils.normalize`).

**Is your enhancement request related to a problem? Please describe.**

I use four functions from `camel_tools.utils` in a service that runs inside a
financial institution's network: `dediac_ar`, `normalize_alef_ar`,
`normalize_alef_maksura_ar` and `normalize_teh_marbuta_ar`. Nothing else from the
toolkit.

Installing camel-tools brings `torch>=2.0` and `transformers>=4.44.0` as hard
dependencies, and on Linux torch's wheel pulls in ~30 `nvidia-*` CUDA packages. My
container image went from 327MB to 3.29GB with camel-tools added, entirely from those
transitive dependencies, for four pure-Python character-map functions.

I understand why torch is required for the BERT disambiguator, NER and sentiment
components. But the `utils` subpackage doesn't need it, and it's the part most likely
to be used in lightweight or constrained environments, where a GPU stack is either
unwelcome (security review) or unaffordable (image size, cold start).

**Describe the solution you'd like**

Make the deep-learning dependencies an install extra, so the normalisation utilities
are installable without torch. Either:

    pip install camel-tools           # utils, tokenizers, morphology — no torch
    pip install "camel-tools[dl]"     # everything, as today

or the inverse, keeping the default full-featured and offering a slim extra. Either
direction works; I'd defer to you on which default is less disruptive for existing
users.

Happy to open a PR for whichever you prefer.

**Describe alternatives you've considered**

For now I've vendored the four functions as `str.maketrans` tables derived from
camel-tools 1.6.0, with the source cited, and kept camel-tools as a test-only
dependency: an equivalence test asserts the vendored functions match upstream on every
codepoint in U+0600–U+06FF, on the whole block as one string, and on each operation
separately. That works, but it means every project with this need re-vendors the same
twenty lines, and drift becomes each project's problem rather than something the
library guarantees.

Also considered `pip install --no-deps` plus manually listing the utils' real
imports, but that's brittle across releases.

**Additional context**

camel-tools 1.6.0, Python 3.12, Linux x86-64 container. The heavy declared
dependencies observed: torch>=2.0, transformers>=4.44.0, scikit-learn>=1.8.0, scipy,
pandas, numpy>=2.0.0, camel-kenlm. The CUDA packages are transitive via torch.

Thanks for the library — the normalisation utilities are exactly right, which is why
I'd like them to be easier to ship.

**Comment by Xcoder2003X** [comment]
could i work on that please ??

**Comment by Molkaat** [comment]
Happy to help with this. 

Looking at the dependency graph: utils, tokenizers, morphology, tagger, cli, data, and disambig.common/disambig.mle have zero heavy imports (torch/transformers/sklearn/scipy). Only disambig.bert, ner, sentiment, and dialectid actually need them. So a pip install camel-tools (light) + pip install "camel-tools[dl]" (adds torch/transformers/scikit-learn/scipy/pandas) split looks clean and matches what @shaniaakhan21 proposed. 

Two things I'd want to handle carefully: 
(1) friendly ModuleNotFoundError messages when someone imports a heavy subpackage without [dl] installed, and 
(2) the camel_diac/camel_morphology CLI entry points, which depend on BERT disambiguation and need to keep failing gracefully on a light install. 

If this is still up for grabs I'd like to take it, happy to open a draft PR to show the approach first.
